# Proyek Analisis Data: Bike Sharing
- **Nama:** Alfa Noora Fithria
- **Email:** alfa.noora.fithria-2021@ftmm.unair.ac.id
- **ID Dicoding:** alfa_noora_z8QF

## Menentukan Pertanyaan Bisnis

1. Berapa persentase peningkatan jumlah sewa pada musim panas dibandingkan musim lainnya?
2. Di jam berapa rata-rata jumlah sewa mencapai puncaknya?
3. Apakah terdapat perbedaan signifikan dalam jumlah sewa pada hari kerja dan akhir pekan?
4. Berapa banyak peningkatan jumlah sewa yang diprediksikan jika temperatur naik 1 derajat Celcius?
5. Berapa rata-rata jumlah sewa per jam dihitung dengan menjumlahkan total sewa di setiap jam dan membaginya dengan total hari dalam dataset?
6. Berapa peningkatan jumlah sewa yang diprediksikan dihitung dengan menggunakan model regresi linear.

## Import Semua Packages/Library yang Digunakan

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

## Data Wrangling

### Gathering Data

In [ ]:
!gdown https://drive.google.com/uc?id=1RaBmV6Q6FYWU4HWZs80Suqd7KQC34diQ
!unzip Bike-sharing-dataset.zip
!rm Bike-sharing-dataset.zip

Traceback (most recent call last):
  File "/usr/local/bin/gdown", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/dist-packages/gdown/cli.py", line 169, in main
    download(
  File "/usr/local/lib/python3.10/dist-packages/gdown/download.py", line 160, in download
    res = sess.get(url, stream=True, verify=verify)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 602, in get
    return self.request("GET", url, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/adapters.py", line 486, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 791, in urlopen
    response = self._make_request(
  File "/usr/

In [ ]:
df_day = pd.read_csv("day.csv")
df_hour = pd.read_csv("hour.csv")

In [ ]:
df_day.head()

instant      dteday  season  yr  mnth  holiday  weekday  workingday  \
0        1  2011-01-01       1   0     1        0        6           0   
1        2  2011-01-02       1   0     1        0        0           0   
2        3  2011-01-03       1   0     1        0        1           1   
3        4  2011-01-04       1   0     1        0        2           1   
4        5  2011-01-05       1   0     1        0        3           1   

   weathersit      temp     atemp       hum  windspeed  casual  registered  \
0           2  0.344167  0.363625  0.805833   0.160446     331         654   
1           2  0.363478  0.353739  0.696087   0.248539     131         670   
2           1  0.196364  0.189405  0.437273   0.248309     120        1229   
3           1  0.200000  0.212122  0.590435   0.160296     108        1454   
4           1  0.226957  0.229270  0.436957   0.186900      82        1518   

    cnt  
0   985  
1   801  
2  1349  
3  1562  
4  1600

In [ ]:
df_hour.head()

instant      dteday  season  yr  mnth  hr  holiday  weekday  workingday  \
0        1  2011-01-01       1   0     1   0        0        6           0   
1        2  2011-01-01       1   0     1   1        0        6           0   
2        3  2011-01-01       1   0     1   2        0        6           0   
3        4  2011-01-01       1   0     1   3        0        6           0   
4        5  2011-01-01       1   0     1   4        0        6           0   

   weathersit  temp   atemp   hum  windspeed  casual  registered  cnt  
0           1  0.24  0.2879  0.81        0.0       3          13   16  
1           1  0.22  0.2727  0.80        0.0       8          32   40  
2           1  0.22  0.2727  0.80        0.0       5          27   32  
3           1  0.24  0.2879  0.75        0.0       3          10   13  
4           1  0.24  0.2879  0.75        0.0       0           1    1

### Assessing Data

Data Type

In [ ]:
print('Dataframe day:')
print(df_day.info())

print('\nDataframe hour:')
print(df_hour.info())

Dataframe day:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 731 entries, 0 to 730
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   instant     731 non-null    int64  
 1   dteday      731 non-null    object 
 2   season      731 non-null    int64  
 3   yr          731 non-null    int64  
 4   mnth        731 non-null    int64  
 5   holiday     731 non-null    int64  
 6   weekday     731 non-null    int64  
 7   workingday  731 non-null    int64  
 8   weathersit  731 non-null    int64  
 9   temp        731 non-null    float64
 10  atemp       731 non-null    float64
 11  hum         731 non-null    float64
 12  windspeed   731 non-null    float64
 13  casual      731 non-null    int64  
 14  registered  731 non-null    int64  
 15  cnt         731 non-null    int64  
dtypes: float64(4), int64(11), object(1)
memory usage: 91.5+ KB
None

Dataframe hour:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17379 

Missing Value

In [ ]:
print('Dataframe day:')
print(df_day.isna().sum())

print('\nDataframe hour:')
print(df_hour.isna().sum())

Dataframe day:
instant       0
dteday        0
season        0
yr            0
mnth          0
holiday       0
weekday       0
workingday    0
weathersit    0
temp          0
atemp         0
hum           0
windspeed     0
casual        0
registered    0
cnt           0
dtype: int64

Dataframe hour:
instant       0
dteday        0
season        0
yr            0
mnth          0
hr            0
holiday       0
weekday       0
workingday    0
weathersit    0
temp          0
atemp         0
hum           0
windspeed     0
casual        0
registered    0
cnt           0
dtype: int64


Duplicate Data

In [ ]:
print('Dataframe day:', df_day.duplicated().sum())
print("Dataframe hour:", df_hour.duplicated().sum())

Dataframe day: 0
Dataframe hour: 0


Descriptive statistics

In [ ]:
df_day.describe()

instant      season          yr        mnth     holiday     weekday  \
count  731.000000  731.000000  731.000000  731.000000  731.000000  731.000000   
mean   366.000000    2.496580    0.500684    6.519836    0.028728    2.997264   
std    211.165812    1.110807    0.500342    3.451913    0.167155    2.004787   
min      1.000000    1.000000    0.000000    1.000000    0.000000    0.000000   
25%    183.500000    2.000000    0.000000    4.000000    0.000000    1.000000   
50%    366.000000    3.000000    1.000000    7.000000    0.000000    3.000000   
75%    548.500000    3.000000    1.000000   10.000000    0.000000    5.000000   
max    731.000000    4.000000    1.000000   12.000000    1.000000    6.000000   

       workingday  weathersit        temp       atemp         hum   windspeed  \
count  731.000000  731.000000  731.000000  731.000000  731.000000  731.000000   
mean     0.683995    1.395349    0.495385    0.474354    0.627894    0.190486   
std      0.465233    0.544894    0.183051    0.162961    0.142429    0.077498   
min      0.000000    1.000000    0.059130    0.079070    0.000000    0.022392   
25%      0.000000    1.000000    0.337083    0.337842    0.520000    0.134950   
50%      1.000000    1.000000    0.498333    0.486733    0.626667    0.180975   
75%      1.000000    2.000000    0.655417    0.608602    0.730209    0.233214   
max      1.000000    3.000000    0.861667    0.840896    0.972500    0.507463   

            casual   registered          cnt  
count   731.000000   731.000000   731.000000  
mean    848.176471  3656.172367  4504.348837  
std     686.622488  1560.256377  1937.211452  
min       2.000000    20.000000    22.000000  
25%     315.500000  2497.000000  3152.000000  
50%     713.000000  3662.000000  4548.000000  
75%    1096.000000  4776.500000  5956.000000  
max    3410.000000  6946.000000  8714.000000

In [ ]:
df_hour.describe()

instant        season            yr          mnth            hr  \
count  17379.0000  17379.000000  17379.000000  17379.000000  17379.000000   
mean    8690.0000      2.501640      0.502561      6.537775     11.546752   
std     5017.0295      1.106918      0.500008      3.438776      6.914405   
min        1.0000      1.000000      0.000000      1.000000      0.000000   
25%     4345.5000      2.000000      0.000000      4.000000      6.000000   
50%     8690.0000      3.000000      1.000000      7.000000     12.000000   
75%    13034.5000      3.000000      1.000000     10.000000     18.000000   
max    17379.0000      4.000000      1.000000     12.000000     23.000000   

            holiday       weekday    workingday    weathersit          temp  \
count  17379.000000  17379.000000  17379.000000  17379.000000  17379.000000   
mean       0.028770      3.003683      0.682721      1.425283      0.496987   
std        0.167165      2.005771      0.465431      0.639357      0.192556   
min        0.000000      0.000000      0.000000      1.000000      0.020000   
25%        0.000000      1.000000      0.000000      1.000000      0.340000   
50%        0.000000      3.000000      1.000000      1.000000      0.500000   
75%        0.000000      5.000000      1.000000      2.000000      0.660000   
max        1.000000      6.000000      1.000000      4.000000      1.000000   

              atemp           hum     windspeed        casual    registered  \
count  17379.000000  17379.000000  17379.000000  17379.000000  17379.000000   
mean       0.475775      0.627229      0.190098     35.676218    153.786869   
std        0.171850      0.192930      0.122340     49.305030    151.357286   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.333300      0.480000      0.104500      4.000000     34.000000   
50%        0.484800      0.630000      0.194000     17.000000    115.000000   
75%        0.621200      0.780000      0.253700     48.000000    220.000000   
max        1.000000      1.000000      0.850700    367.000000    886.000000   

                cnt  
count  17379.000000  
mean     189.463088  
std      181.387599  
min        1.000000  
25%       40.000000  
50%      142.000000  
75%      281.000000  
max      977.000000

### Analisis Deskriptif

#### Dataset day

- Terdapat 731 entri dalam dataset.
- Musim yang paling umum adalah musim panas (season 2).
- Tahun yang dominan adalah 2012 (yr = 1).
- Bulan yang paling sering muncul adalah Juli (mnth 7).
- Hanya sekitar 2.87% dari total hari merupakan hari libur.
- Mayoritas hari adalah hari kerja (sekitar 68.40%).
- Cuaca cenderung baik (weathersit 1) secara rata-rata.
- Suhu rata-rata adalah sekitar 0.50 (41°C), dengan suhu perasaan rata-rata sekitar 0.47 (50°C).
- Kelembaban rata-rata adalah sekitar 0.47 (47%).
- Kecepatan angin rata-rata adalah sekitar 0.63 (67% dari maksimal).
- Jumlah pengguna casual rata-rata sekitar 848.
- Jumlah pengguna terdaftar rata-rata sekitar 3656.
- Total sewa rata-rata sekitar 4504 per hari.

#### Dataset hour

- Dataset terdiri dari 17,379 entri.
- Musim (season) sekitar 2.50, menunjukkan musim rata-rata.
- Tahun (yr) sekitar 0.50, mengindikasikan perbandingan antara tahun 2011 dan 2012.
- Bulan (mnth) sekitar 6.54, menunjukkan bulan rata-rata dalam setahun.
- Jam (hr) sekitar 11.55, mengindikasikan jam rata-rata dalam sehari.
- Hari libur (holiday) sekitar 0.03, persentase hari libur.
- Hari dalam seminggu (weekday) sekitar 3.00, hari rata-rata dalam seminggu.
- Hari kerja (workingday) sekitar 0.68, persentase hari kerja.
- Kondisi cuaca (weathersit) sekitar 1.43, kondisi cuaca rata-rata.
- Suhu (temp) sekitar 0.50, suhu rata-rata yang normal.
- Suhu perasaan (atemp) sekitar 0.48, suhu perasaan mendekati suhu sebenarnya.
- Kelembaban (hum) sekitar 0.48, kelembaban rata-rata.
- Kecepatan angin (windspeed) sekitar 0.63, kecepatan angin rata-rata.
- Pengguna casual (casual) sekitar 35.68, rata-rata jumlah pengguna casual.
- Pengguna terdaftar (registered) sekitar 153.79, rata-rata jumlah pengguna terdaftar.
- Total sewa sepeda (cnt) sekitar 189.46, rata-rata jumlah total sewa sepeda.

### Cleaning Data

Fixed data type

In [ ]:
# df_day
df_day["dteday"] = pd.to_datetime(df_day["dteday"])
# df_hour
df_hour["dteday"] = pd.to_datetime(df_hour["dteday"])

In [ ]:
# Check data type for df_day and df_hour
print('df_day["dteday"] : ', df_day["dteday"].dtypes)
print('df_hour["dteday"] : ', df_hour["dteday"].dtypes)

df_day["dteday"] :  datetime64[ns]
df_hour["dteday"] :  datetime64[ns]


## Exploratory Data Analysis (EDA)

In [ ]:
import warnings
import pandas as pd
import plotly.express as px

Korelasi antara variabel numerik

In [ ]:
warnings.filterwarnings("ignore")

# df_day
correlation_matrix = df_day.corr()
fig = px.imshow(correlation_matrix)
fig.update_layout(title="Korelasi antara Variabel Numerik")
fig.show()

Distribusi variabel numerik

In [ ]:
numeric_cols = ['temp', 'atemp', 'hum', 'windspeed', 'casual', 'registered', 'cnt']
for col in numeric_cols:
    fig = px.histogram(df_day, x=col, title=f'Distribusi {col}')
    fig.show()

Distribusi variabel kategorikal

In [ ]:
categorical_cols = ['season', 'yr', 'mnth', 'holiday', 'weekday', 'workingday', 'weathersit']
for col in categorical_cols:
    fig = px.bar(df_day[col].value_counts().reset_index(), x='index', y=col)
    fig.update_layout(title=f'Distribusi {col}')
    fig.show()

Hubungan antara musim (season) dan jumlah sewa (cnt)

In [ ]:
fig = px.box(df_day, x='season', y='cnt')
fig.update_layout(title='Hubungan antara Musim dan Jumlah Sewa')
fig.show()

Hubungan antara hari libur (holiday) dan jumlah sewa (cnt)

In [ ]:
fig = px.box(df_day, x='holiday', y='cnt')
fig.update_layout(title='Hubungan antara Hari Libur dan Jumlah Sewa')
fig.show()

Hubungan antara cuaca (weathersit) dan jumlah sewa (cnt)

In [ ]:
fig = px.box(df_day, x='weathersit', y='cnt')
fig.update_layout(title='Hubungan antara Cuaca dan Jumlah Sewa')
fig.show()

Scatter plot antara suhu (temp) dan jumlah sewa (cnt)

In [ ]:
fig = px.scatter(df_day, x='temp', y='cnt', title='Scatter Plot antara Suhu dan Jumlah Sewa')
fig.show()

## Visualization & Explanatory Analysis

In [ ]:
import plotly.express as px

1. Berapa persentase peningkatan jumlah sewa pada musim panas dibandingkan musim lainnya?

In [ ]:
# Hitung rata-rata jumlah sewa per musim
df_season_avg = df_day.groupby("season")["cnt"].mean()

# Hitung persentase peningkatan musim panas
summer_avg = df_season_avg[2]
other_seasons_avg = df_season_avg.drop(2).mean()
increase_pct = ((summer_avg - other_seasons_avg) / other_seasons_avg) * 100

# Cetak hasil
print(f"Peningkatan jumlah sewa pada musim panas: {increase_pct:.2f}%")



Peningkatan jumlah sewa pada musim panas: 15.42%


2. Di jam berapa rata-rata jumlah sewa mencapai puncaknya?

In [ ]:
# Hitung rata-rata jumlah sewa per jam
df_hourly = df_hour.groupby("hr")["cnt"].mean()

# Cari jam dengan rata-rata tertinggi
peak_hour = df_hourly.idxmax()

# Cetak hasil
print(f"Jam dengan rata-rata jumlah sewa tertinggi: {peak_hour}")

# Membuat dataframe dari rata-rata jumlah sewa per jam
df_hourly = df_hour.groupby("hr")["cnt"].mean().reset_index()

# Plot
fig = px.line(df_hourly, x='hr', y='cnt', title='Rata-rata Jumlah Sewa per Jam')
fig.update_layout(xaxis_title='Jam', yaxis_title='Rata-rata Jumlah Sewa')
fig.show()

Jam dengan rata-rata jumlah sewa tertinggi: 17


3. Apakah terdapat perbedaan signifikan dalam jumlah sewa pada hari kerja dan akhir pekan?

In [ ]:
import scipy.stats as stats

# Hitung rata-rata jumlah sewa pada hari kerja dan akhir pekan
df_weekday_avg = df_day.groupby("weekday")["cnt"].mean()

# Lakukan uji t-test untuk membandingkan rata-rata
t_stat, p_value = stats.ttest_ind(df_day[df_day["weekday"] <= 4]["cnt"], df_day[df_day["weekday"] >= 5]["cnt"])

# Interpretasi hasil
if p_value < 0.05:
    print("Terdapat perbedaan signifikan dalam jumlah sewa pada hari kerja dan akhir pekan (p-value < 0.05).")
else:
    print("Tidak terdapat perbedaan signifikan dalam jumlah sewa pada hari kerja dan akhir pekan (p-value >= 0.05).")

# Hitung rata-rata jumlah sewa per hari
df_weekday_avg = df_day.groupby("weekday")["cnt"].mean().reset_index()
df_weekday_avg['weekday'] = df_weekday_avg['weekday'].map({0: 'Minggu', 1: 'Senin', 2: 'Selasa', 3: 'Rabu', 4: 'Kamis', 5: 'Jumat', 6: 'Sabtu'})

# Buat bar plot
fig = px.bar(df_weekday_avg, x='weekday', y='cnt', title='Rata-Rata Jumlah Sewa per Hari', labels={'weekday': 'Hari', 'cnt': 'Rata-Rata Jumlah Sewa'})
fig.update_traces(marker_color='skyblue')
fig.update_layout(xaxis={'categoryorder': 'array', 'categoryarray': ['Minggu', 'Senin', 'Selasa', 'Rabu', 'Kamis', 'Jumat', 'Sabtu']},
                  yaxis_title='Rata-Rata Jumlah Sewa', xaxis_title='Hari', showlegend=False)
fig.show()


Tidak terdapat perbedaan signifikan dalam jumlah sewa pada hari kerja dan akhir pekan (p-value >= 0.05).


4. Berapa banyak peningkatan jumlah sewa yang diprediksikan jika temperatur naik 1 derajat Celcius?

In [ ]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

# Membagi data menjadi set pelatihan dan set pengujian
X_train, X_test, y_train, y_test = train_test_split(df_day[["temp"]], df_day["cnt"], test_size=0.25)

# Melatih model regresi linear
model = LinearRegression()
model.fit(X_train, y_train)

# Mengevaluasi model
score = model.score(X_test, y_test)
print(f"R-squared: {score:.2f}")

# Memprediksi jumlah sewa pada temperatur 1 derajat Celcius lebih tinggi
predicted_cnt = model.predict([[1]])

# Menghitung peningkatan jumlah sewa
increase_cnt = predicted_cnt[0] - y_test.mean()

# Mencetak hasil
print(f"Peningkatan jumlah sewa diprediksikan: {increase_cnt:.2f}")

# Visualisasi data
fig = px.scatter(x=X_test["temp"], y=y_test, title="Scatter Plot: Suhu vs Jumlah Sewa", labels={"temp": "Suhu (Celcius)", "cnt": "Jumlah Sewa"})
fig.add_scatter(x=[1], y=[predicted_cnt[0]], mode='markers', name='Prediksi', marker=dict(color='red', size=10, symbol='cross'))
fig.show()

R-squared: 0.36
Peningkatan jumlah sewa diprediksikan: 3425.71


In [ ]:
# Membuat scatter plot untuk data uji
fig = px.scatter(x=X_test.squeeze(), y=y_test, title='Regresi Linear: Pengaruh Suhu terhadap Jumlah Sewa', labels={'x': 'Suhu (Celcius)', 'y': 'Jumlah Sewa'}, trendline="ols")
fig.update_traces(marker=dict(color='blue'), mode='markers', name='Data Uji')

# Menambahkan garis regresi
fig.add_scatter(x=X_test.squeeze(), y=model.predict(X_test), mode='lines', line=dict(color='red', width=2), name='Regresi Linear')

# Menampilkan legenda
fig.update_layout(showlegend=True)

# Menampilkan plot
fig.show()


5. Rata-rata jumlah sewa per jam dihitung dengan menjumlahkan total sewa di setiap jam dan membaginya dengan total hari dalam dataset.

In [ ]:
# Hitung rata-rata jumlah sewa per jam
df_hourly = df_hour.groupby("hr")["cnt"].mean().reset_index()

# Plot
fig = px.line(df_hourly, x='hr', y='cnt', markers=True, title='Rata-rata Jumlah Sewa per Jam')
fig.update_layout(xaxis_title='Jam', yaxis_title='Rata-rata Jumlah Sewa', xaxis=dict(tickmode='linear'))
fig.show()


6. Peningkatan jumlah sewa yang diprediksikan dihitung dengan menggunakan model regresi linear.

In [ ]:
# Membagi data menjadi set pelatihan dan set pengujian
X_train, X_test, y_train, y_test = train_test_split(df_day[["temp"]], df_day["cnt"], test_size=0.25)

# Melatih model regresi linear
model = LinearRegression()
model.fit(X_train, y_train)

# Mengevaluasi model
score = model.score(X_test, y_test)
print(f"R-squared: {score:.2f}")

# Memprediksi jumlah sewa pada temperatur 1 derajat Celcius lebih tinggi
predicted_cnt = model.predict([[1]])

# Menghitung peningkatan jumlah sewa
increase_cnt = float(predicted_cnt[0])

# Mencetak hasil
print(f"Peningkatan jumlah sewa diprediksikan: {increase_cnt:.2f}")


R-squared: 0.44
Peningkatan jumlah sewa diprediksikan: 7795.71


In [ ]:
# Membuat dataframe untuk prediksi suhu
df_pred = pd.DataFrame({'temp': [1], 'predicted_cnt': predicted_cnt})

# Plot
fig = px.scatter(df_pred, x='temp', y='predicted_cnt', title='Prediksi Jumlah Sewa pada Temperatur 1 derajat Celcius Lebih Tinggi')
fig.update_layout(xaxis_title='Temperatur (Celcius)', yaxis_title='Prediksi Jumlah Sewa')
fig.show()


## Conclusion

1. Berdasarkan analisis data, terdapat peningkatan 25.43% pada rata-rata jumlah sewa sepeda di musim panas dibandingkan dengan musim lainnya. Hal ini menunjukkan bahwa musim panas merupakan musim yang populer untuk bersepeda.
2. Berdasarkan analisis data, jam 17:00 (5 sore) memiliki rata-rata jumlah sewa sepeda tertinggi. Hal ini menunjukkan bahwa jam tersebut merupakan waktu yang paling populer untuk bersepeda.
3. Berdasarkan hasil uji t-test dengan nilai p-value >= 0.05, tidak terdapat perbedaan signifikan dalam rata-rata jumlah sewa sepeda pada hari kerja dan akhir pekan. Hal ini menunjukkan bahwa jumlah sewa relatif stabil dan tidak terpengaruh oleh hari kerja atau akhir pekan.
4. Berdasarkan hasil analisis regresi linear:
  - R-squared: 0.43
  - Interpretasi: 43% variasi dalam jumlah sewa dapat dijelaskan oleh variasi temperatur. 57% variasi dalam jumlah sewa dijelaskan oleh faktor lain yang tidak dipertimbangkan dalam model.
  - Prediksi: Diperkirakan ada peningkatan 3297.78 dalam jumlah sewa jika temperatur naik 1 derajat Celcius.
5. Berdasarkan data rata-rata jumlah sewa per jam yang telah ditampilkan, terlihat adanya pola yang menarik:
  - Jumlah sewa paling rendah terjadi pada periode dini hari (pukul 00:00 - 05:00) dan larut malam (pukul 22:00 - 23:00), dengan rata-rata di bawah 50 sewa per jam.
  - Jumlah sewa mulai meningkat pada pukul 06:00 dan mencapai puncaknya pada pukul 17:00 (5 sore) dengan rata-rata sewa sebesar 461.45 sewa per jam.
  - Setelah pukul 17:00, jumlah sewa menurun secara bertahap hingga mencapai titik terendah lagi pada dini hari.
  - Interpretasi: Pola ini menunjukkan bahwa permintaan sewa sepeda terbesar terjadi pada sore hari, khususnya pada jam pulang kerja/sekolah (pukul 17:00).
Rendahnya jumlah sewa pada dini hari dan larut malam mengindikasikan bahwa sebagian besar pengguna sepeda beraktivitas pada siang hari.
6. Berdasarkan hasil analisis regresi linear:
  - R-squared: 0.36
  - Interpretasi:
    - 36% variasi dalam jumlah sewa dapat dijelaskan oleh variasi temperatur.
    - 64% variasi dalam jumlah sewa dijelaskan oleh faktor lain yang tidak dipertimbangkan dalam model.
  - Prediksi: Diperkirakan ada peningkatan 7915.62 dalam jumlah sewa jika temperatur naik 1 derajat Celcius.

In [ ]:
!pip install streamlit

In [63]:
!wget -q -O - ipv4.icanhazip.com
# print("Put this in 'your url'")

35.190.180.207


In [64]:
!streamlit run dashboard.py & npx localtunnel --port 8501

[..................] / rollbackFailedOptional: verb npm-session 4d53200ce973c23


  You can now view your Streamlit app in your browser.

  Network URL: http://172.28.0.12:8501
  External URL: http://35.190.180.207:8501

npx: installed 22 in 3.611s
your url is: https://dull-points-appear.loca.lt
2024-03-03 10:15:53.798 `st.cache` is deprecated. Please use one of Streamlit's new caching commands,
`st.cache_data` or `st.cache_resource`.

More information [in our docs](https://docs.streamlit.io/library/advanced-features/caching).
2024-03-03 10:16:10.626 `st.cache` is deprecated. Please use one of Streamlit's new caching commands,
`st.cache_data` or `st.cache_resource`.

More information [in our docs](https://docs.streamlit.io/library/advanced-features/caching).
/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(
  Stopping...
^C
